### PR TITLE
[risk=low][RW-9200] Upgrade React Types

### PR DESCRIPTION
**Pull request by jmthibault79**
Upgrade react-types and react-types-dom to match our version of React.

No expected changes.  Tested locally.

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [x] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [x] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/components/genomics-extraction-table.tsx [hunk 1]
```diff
@@ -207,7 +207,7 @@ const FailedRequestMessage = () => (
 );
 
 export const GenomicsExtractionTable = fp.flow(withCurrentWorkspace())(
-  ({ workspace }) => {
+  ({ workspace }: WithCurrentWorkspace) => {
```
Comment (jmthibault79) [inline ui/src/app/components/genomics-extraction-table.tsx:210]: not really needed here, just happened to see it

File: ui/src/app/components/popups.tsx [hunk 2]
```diff
@@ -59,8 +59,8 @@ interface WithDynamicPositionProps {
   handleClickOutside?: () => void;
   outsideClickIgnoreClass?: string;
   onClick?: () => void;
+  children?: React.ReactNode;
```
Comment (jmthibault79) [inline ui/src/app/components/popups.tsx:62]: This is always called with children, and always passes its remaining props to the wrapped component.

The compiler started complaining, so here is a fix.

File: ui/src/app/pages/data/cohort/demographics.tsx [hunk 3]
```diff
@@ -269,11 +269,8 @@ export class Demographics extends React.Component<Props, State> {
     this.setState({ ageType }, () => this.calculateAgeFromNodes());
   }
 
-  onSliderInit(slider: React.ReactNode) {
```
Comment (jmthibault79) [inline ui/src/app/pages/data/cohort/demographics.tsx:272]: now there's a second problem: it's also not a ReactNode!  The easiest solution seems to be to keep this untyped

Comment (jmthibault79) [reply ui/src/app/pages/data/cohort/demographics.tsx:272]: Tested this: Cohort -> Demographics -> Age

The age slider and count have the same behavior as before.

**Review by evrii — APPROVED** [review]
Ran locally with no clear errors. Thanks for taking care of this.

**Pull request merged**